src/boot/OfflineNoticeProvider.js

```diff
@@ -0,0 +1,454 @@
+// @flow strict-local
+import React, { useCallback, useContext, useEffect, useMemo, useRef, useState } from 'react';
+import type { Node } from 'react';
+import {
+  AccessibilityInfo,
+  useWindowDimensions,
+  View,
+  Animated,
+  LayoutAnimation,
+  Platform,
+  Easing,
+} from 'react-native';
+import NetInfo from '@react-native-community/netinfo';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import type { ViewProps } from 'react-native/Libraries/Components/View/ViewPropTypes';
+import type { DimensionValue } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';
+
+import * as logging from '../utils/logging';
+import { useDispatch, useGlobalSelector } from '../react-redux';
+import { getGlobalSession } from '../directSelectors';
+import { useConditionalEffect, useHasStayedTrueForMs, usePrevious } from '../reactUtils';
+import type { JSONableDict } from '../utils/jsonable';
+import { createStyleSheet } from '../styles';
+import ZulipTextIntl from '../common/ZulipTextIntl';
+import { HALF_COLOR } from '../styles/constants';
+import type { ViewStylePropWithout } from '../reactUtils';
+import ZulipStatusBar from '../common/ZulipStatusBar';
+import { TranslationContext } from './TranslationProvider';
+import { appOnline } from '../session/sessionActions';
+
+function useUpdateSessionOnConnectivityChange() {
```

Comment:
I considered this, and it would be a nice simplification to the current code. But I haven't gone through with it yet, because I figure we still have work to do toward an intentional "offline mode", like defining and implementing a reduced feature set to offer in that mode. For that, I expect it to be handy to have a piece of global state like this, though I'm not sure if it should be our `isOnline` exactly as-is. With these thoughts in mind, this simplification seemed out-of-scope for this PR.

Reply:
Sounds good; that's a fine reason to put off that refactor and see what develops.